dbt_semantic_interfaces/implementations/metric.py

```diff
@@ -180,6 +180,7 @@ class PydanticMetric(HashableBaseModel, ModelWithMetadataParsing):
     filter: Optional[PydanticWhereFilterIntersection]
     metadata: Optional[PydanticMetadata]
     label: Optional[str] = None
+    meta: Dict[str, Any] = {}
```

Comment:
We should move this under a `config` property as we have done in `SemanticModel`. We should do this for two reasons
1. Consistency across the protocols (meta should always be found in the same place)
2. In dbt-core the direct `meta` attribute is deprecated in favor of `meta` being under the `config` property
  a. In accordance with this, we have an [open issue](https://github.com/dbt-labs/dbt-core/issues/9441) which we plan to take on shortly to bring the core definition of `Metric` into alignment 
  
Additionally we should update the default setting to
```
meta: Dict[str, Any] = Field(default_factory=dict)
```
as described in another comment